### DISCUSSION TITLE
APK For Chitchatter

### DISCUSSION THREAD

**Discussion by MohammadTaseenKhan** [jeremyckahn/chitchatter#37] [Show and tell]
I have created an apk for Chitchatter.im 

I want to share it here for testing purpose.

Click Here to download 

[Download APK](https://taseen-server.pages.dev/Apk/Chitchatter.apk)

```bash
Link preview for safety check: https://taseen-server.pages.dev/Apk/Chitchatter.apk
```

**Comment by jeremyckahn** [comment]
Whoa! This is very cool. Thank you for sharing this @MohammadTaseenKhan! I don't have an Android phone, so I'll need to set up a development environment to test this out. I will do that when I have the time.

How did you produce this APK? If it can be automated, we could potentially provide official Chitchatter APKs via GitHub Actions. As long as it's transparent how the binary was produced (via public GitHub Action logs), this could be a great way to reach more users.

**Comment by MohammadTaseenKhan** [comment]
First of all I was trying to create an apk so that i do not have to go back to the browser to use chitchatter 
Just one click will take me to the web. I have used an embedding web system , Is like i have embed your web platform into the apk  so that in every android it will get responsive to its layout.

Chitchatter downloads some of its data in the background of the browser.
The web app will do the same but little different it will download the some of its data in the background of the Application not the browser. 

As like it will have no connection with the browser. 

Because we know some of its cookies web data can cause a great problem to deal with anonymous secure chatting.

> This apk is safe because there is no tracking link added to it

> No external binary has been taken from some where like in any random website.

> Used APK Tool to create this apk 

[APK Tool](https://www.kali.org/tools/apktool/)

**Comment by jeremyckahn** [comment]
This is a really interesting tool. Thanks for sharing this @MohammadTaseenKhan! I'm wondering how this might compare to https://github.com/GoogleChromeLabs/bubblewrap. Bubblewrap converts a Progressive Web App (such as Chitchatter) to an Android app.

Regardless of the tool used, I think it would be great to have a GitHub Action that produces a APK file. Such an APK could be distributed on the Google Play store, but I'd also like to provide it here on GitHub so that users can be sure of the authenticity of the binary.

I don't have any plans to implement this in the near future myself, but I would happy to work with anyone who would like to submit a PR to get it done!
